### DISCUSSION TITLE
Quality of Life / Accessibility Features

### DISCUSSION THREAD

**Discussion by therocco** [sunflower-land/sunflower-land#2348] [Ideas]
With the recent release of more quests I wanted to suggest some quality of life improvements that would also have the extra added benefit of making the game more accessible to all users. While the new quests are wonderful they do require a lot of jumping back and fourth between your island and various other islands where the quest keeper is located (e.g. Helio for Hayseed Hank quests). You also have to visit the person that gave you the quest if you take a break or get distracted and forget what you have to harvest, eat, dig, etc.

I would like to propose that we turn the current Bumpkin menu that you see when you click your Bumpkin in the top left hand corner into a tabbed modal menu that also lets you interact with a quest menu. In the quest menu it can let you know what the current task is for all active quests and let you complete them from that menu if need be and even possible see and start the next quest in the quest line only after you complete the current quest.

I would also like to propose adding in features to make the game more accessible such as adding `alt` tags to images and `tab-index` attributes to interactive elements so users that require assistive technology like screen readers or users that have a difficult time using a mouse and keyboard on desktop can enjoy an easy to use pain free experience that able bodied players experience.

As part of making things more accessible I also propose adding in more keyboard shortcuts to make it easier for players to navigate throughout the game with fewer mouse clicks. Attached is an example screenshot of some keyboard shortcuts that could be used to activate the different HUD elements in the game. Not all users have the dexterity to efficiently use a mouse type input device so shortcuts like these can be extremely beneficial.

![Screenshot from 2023-03-19 00-10-11](https://user-images.githubusercontent.com/801056/226160200-6acf8590-ba98-4dbd-a6cc-95b476ca016b.png)

**Comment by brian-man** [comment]
> I would also like to propose adding in features to make the game more accessible such as adding alt tags to images and tab-index attributes to interactive elements so users that require assistive technology like screen readers or users that have a difficult time using a mouse and keyboard on desktop can enjoy an easy to use pain free experience that able bodied players experience.

> As part of making things more accessible I also propose adding in more keyboard shortcuts to make it easier for players to navigate throughout the game with fewer mouse clicks. Attached is an example screenshot of some keyboard shortcuts that could be used to activate the different HUD elements in the game. Not all users have the dexterity to efficiently use a mouse type input device so shortcuts like these can be extremely beneficial.

Thanks for raising this post and especially these two ideas.  In addition to opening the game up to broader appeal and audiences, this is kind of "table stakes" for a game that is aiming to be "human-friendly".

**Comment by popalexandrudan** [comment]
also please, remove triple click for nodes. instead use double click. i don't see any reason it should be 3x instead of 2x

**Comment by therocco** [comment]
An accessibility setting also enable users to use the spacebar for harvesting and enter keys for buildings/npcs if a user is tabbing between interactive items would be helpful. So instead of multiple 2-3x mouse clicks on a tree you could double tap the spacebar
